src/otx/v2/adapters/torch/mmengine/mmdet/engine.py

```diff
@@ -0,0 +1,214 @@
+"""OTX adapters.torch.mmengine.mmdet.Engine API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import torch
+from mmdet.registry import VISUALIZERS
+
+from otx.v2.adapters.torch.mmengine.engine import MMXEngine
+from otx.v2.adapters.torch.mmengine.mmdet.registry import MMDetRegistry
+from otx.v2.adapters.torch.mmengine.modules.utils.config_utils import CustomConfig as Config
+from otx.v2.api.utils.logger import get_logger
+
+if TYPE_CHECKING:
+    import numpy as np
+
+logger = get_logger()
+
+
+class MMDetEngine(MMXEngine):
+    """The MMDetEngine class is responsible for running inference on pre-trained models."""
+
+    def __init__(
+        self,
+        work_dir: str | Path | None = None,
+    ) -> None:
+        """Initialize a new instance of the MMDetEngine class.
+
+        Args:
+            work_dir (Optional[Union[str, Path]], optional): The working directory for the engine. Defaults to None.
+            config (Optional[Union[Dict, Config, str]], optional): The configuration for the engine. Defaults to None.
+        """
+        super().__init__(work_dir=work_dir)
+        self.registry = MMDetRegistry()
+
+    def _update_config(self, func_args: dict, **kwargs) -> tuple[Config, bool]:
```

Comment:
can't we reuse the method defined in mmengine?
When it comes to me, some configs such as val/test evaluators, visualizer, or param scheduler could be.

Reply:
When engines are used by api user, and that user do not provide proper func_args, engine load default func_args from https://github.com/openvinotoolkit/training_extensions/blob/e6c7325c506e93dec7c1b6a5ab9e63bc80a9d50c/src/otx/v2/adapters/torch/mmengine/engine.py#L36
However, default config needs some additional patches to use in each algo engine. Therefore this additional codes are added. I agree this should be removed in next phase. What do you think?

Reply:
alright, thank you!